### PR TITLE
fix(anta.tests): Fix VerifyNTPAssociations to support IP-DNS peers

### LINKED ISSUE #900: Bug: False positive on VerifyNTPAssociation
`show ntp associations | json` can return the following output:
```
DC05-Temp-Bleaf1#sho ntp associations | json
{
    "peers": {
        "10.10.200.22 (ntp.example.com)": {
            "condition": "sys.peer",
            "peerIpAddr": "10.10.200.22",
            "refid": "164.100.255.122",
            "stratumLevel": 3,
            "peerType": "unicast",
            "lastReceived": 1730124132.0,
            "pollInterval": 1024,
            "reachabilityHistory": [
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true
            ],
            "delay": 0.524,
            "offset": 5.545,
            "jitter": 3.8
        }
    }
}
```
The FQDN of the NTP server can be included in parentheses. The test logic should be updated accordingly.

### PR DESCRIPTION
**Pull request by carl-baillargeon**
# Description

Update VerifyNTPAssociations test to support IP-DNS peers in the JSON output. We now check if the provided NTP server IP or DNS is matching either the peer key or `peerIpAddr` in the details.

Fixes #900 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: tests/units/anta_tests/test_system.py [hunk 1]
```diff
@@ -346,6 +346,39 @@
         },
         "expected": {"result": "success"},
     },
+    {
+        "name": "success-ip-dns",
+        "test": VerifyNTPAssociations,
+        "eos_data": [
+            {
+                "peers": {
+                    "1.1.1.1 (1.ntp.networks.com)": {
```
Comment (MaheshGSLAB) [inline tests/units/anta_tests/test_system.py:355]: test will still pass if we have actual address as 1.1.1.11 and input address as 1.1.1.1

Comment (MaheshGSLAB) [reply tests/units/anta_tests/test_system.py:355]: Better if we can check the input address with peerIpAddr

Comment (carl-baillargeon) [reply tests/units/anta_tests/test_system.py:355]: Good catch. Will fix it.

Comment (carl-baillargeon) [reply tests/units/anta_tests/test_system.py:355]: Fixed in c8928bd0085d3e01d5fe77ca99065c22d71cae83

**Review by gmuloc — APPROVED** [review]

**Pull request merged**